src/storage/benches/bench_compactor.rs

```diff
@@ -0,0 +1,299 @@
+        sstable_id,
+        SSTableBuilderOptions {
+            capacity: 32 * 1024 * 1024,
+            block_capacity: 16 * 1024,
+            restart_interval: 16,
+            bloom_false_positive: 0.01,
+            compression_algorithm: CompressionAlgorithm::None,
+        },
+    );
+    let value = b"1234567890123456789";
+    let mut full_key = test_key_of(0, epoch);
+    let user_len = full_key.len() - 8;
+    for i in range {
+        let start = (i % 8) as usize;
+        let end = (start + 8) as usize;
+        full_key[(user_len - 8)..user_len].copy_from_slice(&i.to_be_bytes());
+        builder.add(&full_key, HummockValue::put(&value[start..end]));
+    }
+    let (_, data, meta, _) = builder.finish();
+    (data, meta)
+}
+
+async fn scan_all_table(sstable_store: SstableStoreRef) {
+    let mut stats = StoreLocalStatistic::default();
+    let table = sstable_store.sstable(1, &mut stats).await.unwrap();
+    let default_read_options = Arc::new(ReadOptions::default());
+    // warm up to make them all in memory. I do not use CachePolicy::Fill because it will fetch
+    // block from meta.
+    let mut iter = SSTableIterator::new(table, sstable_store.clone(), default_read_options);
+    iter.rewind().await.unwrap();
+    while iter.is_valid() {
+        iter.next().await.unwrap();
+    }
+}
+
+async fn scan_all_table_inline(sstable_store: SstableStoreRef) {
+    let mut stats = StoreLocalStatistic::default();
+    let table = sstable_store.sstable(1, &mut stats).await.unwrap();
+    let default_read_options = Arc::new(ReadOptions::default());
+    // warm up to make them all in memory. I do not use CachePolicy::Fill because it will fetch
+    // block from meta.
+    let mut iter = SSTableIterator::new(table, sstable_store.clone(), default_read_options);
+    iter.rewind().await.unwrap();
+    while iter.is_valid() {
+        iter.next_inner().await.unwrap();
+    }
+}
+
+fn bench_table_build(c: &mut Criterion) {
+    c.bench_function("bench_table_build", |b| {
+        b.iter(|| {
+            let _ = build_table(0, 0..(MAX_KEY_COUNT as u64), 1);
+        });
+    });
+}
+
+fn bench_table_scan(c: &mut Criterion) {
+    let (data, meta) = build_table(0, 0..(MAX_KEY_COUNT as u64), 1);
+    let sstable_store = mock_sstable_store();
+    let runtime = tokio::runtime::Builder::new_current_thread()
+        .build()
+        .unwrap();
+    let sstable_store1 = sstable_store.clone();
+    runtime.block_on(async move {
+        sstable_store1
+            .put(Sstable::new(1, meta.clone()), data, CachePolicy::NotFill)
+            .await
+            .unwrap();
+    });
+    // warm up to make them all in memory. I do not use CachePolicy::Fill because it will fetch
+    // block from meta.
+    let sstable_store1 = sstable_store.clone();
+    runtime.block_on(async move {
+        scan_all_table(sstable_store1).await;
+    });
+
+    c.bench_function("bench_table_iterator", |b| {
+        b.iter(|| {
```

Comment:
criterion has support for async executors (See https://bheisler.github.io/criterion.rs/book/user_guide/benchmarking_async.html). Consider using it?

Reply:
done